app/hotels/src/components/filters/FilterButton.js

```diff
@@ -0,0 +1,56 @@
+// @flow
+
+import * as React from 'react';
+import idx from 'idx';
+import { View } from 'react-native';
+import { connect } from 'react-redux';
+import { Button, Color } from '@kiwicom/native-common';
+
+const styles = {
+  default: {
+    padding: 3,
+    paddingHorizontal: 3,
+  },
+};
+
+type Props = {
+  title: string,
+  filter: string,
+  isActive: boolean,
+  onPress: (filter: string) => void,
+};
+
+export const UnwrappedFilterButton = (props: Props) => {
+  const backgroundColor = props.isActive ? Color.grey._900 : Color.brand;
+  return (
+    <View style={styles.default}>
+      <Button
+        title={props.title}
+        onPress={() => props.onPress(props.filter)}
+        icon={{
+          name: 'attach-money',
+        }}
+        backgroundColor={backgroundColor}
+      />
+    </View>
+  );
+};
+
+const mapStateToProps = (state, props) => {
+  return {
+    isActive:
+      idx(state.hotelsSearch.filters, _ => _[props.filter].isActive) || false,
+  };
+};
+const mapDispatchToProps = dispatch => ({
+  onPress: (filter: string) => {
+    dispatch({
+      type: 'toggleFilter',
+      filter,
+    });
+  },
+});
+
+export default connect(mapStateToProps, mapDispatchToProps)(
+  UnwrappedFilterButton,
+);
```

Comment:
It seems really weird to me that simple on/off button needs Redux. Is this really problem of the button? Shouldn't this be problem of the filter? Why do you need Redux after all?